crates/rspack_core/src/module_graph.rs

```diff
@@ -125,7 +124,7 @@ impl ModuleGraph {
   }
 
   pub fn module_identifier_by_dependency_id(&self, id: &DependencyId) -> Option<&ModuleIdentifier> {
-    self.dependency_id_to_module_identifier.get(id)
+    self.dependency_id_to_module_identifier[*id].as_ref()
```

Comment:
All `[]` index access are safe because we do append only operations. We can refactor the backing `Vec` to a newtype if we want to enforce this. But I'll leave this to the next person.